### PR TITLE
Topic recognition

### PR DESCRIPTION
**Pull request by jackCashman03**
This pull request contains all relevant files and images detailing my implementation of a generative VQVAE with PixelCNN prior to act as a generative model for the ADNI Brain dataset. The model produces a clear image and meets criteria of mean SSIM greater than 0.6

Files included are:

modules.py : source code of the components of your model.
dataset.py : data loader for loading and preprocessing your data
predict.py : example usage of trained model
train.py : source code for training / validating / testing / saving model.
README.md : Comprehensive documentation
./Images : A collection of images detailing results from testing.

Model achieved 0.9257 / 0.92562 Mean SSIM on validation / test set on 7 epochs with hyper-parameters specified at the top of train.py. Fore more detailed outline, see README.md. All in all, this was a very hard, but enjoyable project! (Though I didn't enjoy the Rangpur queue times lol)

Thanks,
Jack Cashman (Student number 47431748)

### CODE REVIEW

**Comment by SiyuLiu0329** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme Overall:  Good
- Project Overview: Good
- Model: Good
- Data + Preprocessing: Good
- Usage: Good
- Training / Loss Curve(s): Good
- Result Demonstration: Good
- References: Good

Functionality:
- Reconstruction: Good
- Generation: No results shown

Code:
- Consistent with Results: Yes
- File Structure: Good
- Commenting: Good
- Commit frequency: Good
- Commit messages: Good

Other Comments:
- I can see pixel CNN training and inference code, but no results have been given.

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage, no generation results -1
Module present
Commenting
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable/good
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good
